File: DesignPatternsJava/src/com/decorator/Leminate.java
package com.decorator;
//Lemination costs 20 taka extra
public class Leminate extends BookMaker {
	public Leminate(IBook book) {
		super(book);
		System.out.println("Leminated");
	}
	public String getDescription() {
		return aBook.getDescription()+", Leminated";
	}
	public double getPrice() {
		return aBook.getPrice()+20.00;
	}
}

File: DesignPatternsDotNet/Decorator/Leminate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Decorator
{   
    public class Leminate : BookMaker
    {
        public Leminate(IBook book)
            : base(book)
        {
            Console.WriteLine("Leminate covering");
        }
        public string Description { get { return aBook.Description + ", Leminateting"; } set { aBook.Description = value; } }
        public override double Price { get { return aBook.Price + 20.00; } set { aBook.Price = value; } }
    }    
}
